[CIAPP] Document dd_metrics for buildkite pipelines (#13168)
DataDog/documentation: content/en/continuous_integration/setup_pipelines/buildkite.md

### Setting custom tags

Custom tags can be added to Buildkite traces by using the `buildkite-agent meta-data set` command.

> Any metadata tags with a key starting with `dd_tags.` are added to the job and pipeline spans. This
> tags can be used to create string facets to search and organize the pipelines.

Reviewer: ```suggestion
Any metadata tags with a key starting with `dd_tags.` are added to the job and pipeline spans. These tags can be used to create string facets to search and organize the pipelines.
```

{{< img src="ci/buildkite-custom-tags.png" alt="Buildkite pipeline trace with custom tags" style="width:100%;">}}

To create numerical tags that can be use for measures the metadata command can be used. Any metadata
with a key starting with `dd_metrics.` and a numerical value will be set as a metric tag.

> This can be used for example to set the binary size on the pipeline:

Reviewer 2: ```suggestion
This can be used for example to measure the binary size in a pipeline:
```

> To create numerical tags that can be use for measures the metadata command can be used. Any metadata
> with a key starting with `dd_metrics.` and a numerical value will be set as a metric tag.
> This can be used for example to measure the binary size in a pipeline:

Reviewer: ```suggestion
Any metadata with a key starting with `dd_metrics.` and containing a numerical value will be set as a metric tag.

You can use the `buildkite-agent meta-data set` command to create tags with numerical values. Any metadata with a key starting with `dd_metrics.` and a numerical value will be set as a metric tag. The following example measures the binary size in a pipeline:
```

```yaml
steps:
  - commands:
    - go build -o dst/binary .
    - ls -l dst/binary | awk '{print \$5}' | tr -d '\n' | buildkite-agent meta-data set "dd_metrics.binary_size"
    label: Go build
```

The resulting pipeline will have the tags shown below in the pipeline span:

- `binary_size: 502` (output depends on the file size)

> This value can be used to plot the change in the binary size over time for example.

Reviewer: ```suggestion
In this example, you can use the value of `binary_size` to plot the change in the binary size over time.
```
